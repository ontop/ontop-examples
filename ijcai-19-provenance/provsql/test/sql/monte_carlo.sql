\set ECHO none
SET search_path TO public, provsql;

CREATE TABLE mc_result AS
SELECT city, probability_evaluate(provenance(),'p','monte-carlo','10000') AS prob
FROM (
  SELECT DISTINCT city
  FROM personnel
EXCEPT 
  SELECT p1.city
  FROM personnel p1,personnel p2
  WHERE p1.id<p2.id AND p1.city=p2.city
  GROUP BY p1.city
) t;

SELECT remove_provenance('mc_result');

SELECT city, ROUND(prob::numeric,1) FROM mc_result WHERE city = 'Paris';
DROP TABLE mc_result;
