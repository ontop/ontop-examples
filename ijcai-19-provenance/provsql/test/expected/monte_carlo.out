\set ECHO none
 remove_provenance 
-------------------
 
(1 row)

 city  | round 
-------+-------
 Paris |   0.4
(1 row)

